src/main/groovy/jenkins/automation/builders/MultibranchPipelineJobBuilder.groovy

```diff
@@ -1,55 +1,85 @@
 package jenkins.automation.builders
 
-import jenkins.automation.utils.CommonUtils
-import jenkins.automation.utils.Branch
-
 import javaposse.jobdsl.dsl.DslFactory
 import javaposse.jobdsl.dsl.jobs.MultibranchWorkflowJob
-import jenkins.automation.utils.Branch
+
+
+/*
+    NB: if you are polling via the GitHub API (github.com or GHE) and
+        want to use webhooks with a multi-branch pipeline, but the webhooks
+        are not being created, it may be because you need to update Jenkins
+        plugins; ensure that the 'Branch API Plugin' and the 'GitHub Branch
+        Source Plugins' are both up to date
+*/
+
 
 /**
- * Multibranch Pipeline Job builder creates a set of Pipeline projects according to detected branches in a Git repository
+ * Multibranch Pipeline Job builder creates a set of Pipeline projects
+ * according to detected branches in a Git repository
  * @param name job name
  * @param description description for the folder
- * @param branchSource set to "Github" for Github enterprise or "Git" for public Github repos
- * @param gitCredentials credentials used to scan branches
- * @param gitOwner name of the GitHub Organization or GitHub User Account when branchSource is "Github"
- * @param gitRepository name of the GitHub repository when branchSource is "Github"
- * @param gitEndpoint GitHub API endpoint when branchSource is "Github"
- * @param gitRemote Git remote project repository URL when branchSource is "Git"
- * @param oldNumToKeep (optional) Number of builds to keep after a git branch has been removed
+ * @param branchSource where the git repo lives; see <code>BranchSourceType</code>
+ * for options
+ * @param gitCredentials Jenkins git credentials used to scan branches
+ * @param ghOwner name of the organization or user when branchSource is 'GitHub'
+ * @param ghRepo name of the git repo when branchSource is 'GitHub'
+ * @param ghApiEndpoint (optional) the URL of the GitHub API endpoint
+ * when branchSource is 'GitHub'; when pointing to a GitHub Enterprise API URL,
+ * you will first need to make the corresponding configurations in your Jenkins's
+ * 'GitHub' and 'GitHub Enterprise Servers' sections on the 'Configure System'
+ * page (otherwise a custom non-github.com API endpoint will not work)
+ * @param gitRemote the URL of the git repo when branchSource is 'git'
+ * @param oldNumToKeep (optional) Number of builds to keep after a git branch
+ * has been removed
  */
 class MultibranchPipelineJobBuilder {
     String name
     String description
-    String branchSource
+    BranchSourceType branchSource
     String gitCredentials
-    String gitOwner
-    String gitRepository
-    String gitEndpoint
+    String ghOwner
+    String ghRepo
+    String ghApiEndpoint
     String gitRemote
     int oldNumToKeep = 10
 
+    enum BranchSourceType {
+        /**
```

Comment:
Style choice on multi-line here?

Reply:
good question; doing it that way was the only way I could get these comments to be included as `description` in the groovydoc

Reply:
This was not a pun, I swear.